### PR TITLE
Fix missing sort of set_intersection parameters

### PR DESCRIPTION
**Pull request by orhan-mehmedov**
Added sorting of std::set_intersection parameters.
Fixed in cache_manager.cc.
Fixes assert in windows platform.

Related to [APPLINK-27511](https://adc.luxoft.com/jira/browse/APPLINK-27511).

### CODE REVIEW

**Comment by orhan-mehmedov** [comment]
Please review,
@AByzhynar @LuxoftAKutsan @anosach-luxoft @dev-gh @Kozoriz @wolfylambova22 @pvvasilev

**Comment by dev-gh** [comment]
Reviewed

**Comment by pvvasilev** [comment]
@orhan-mehmedov Reviewed.

**Comment by wolfylambova22** [comment]
@orhan-mehmedov Reviewed.

**Comment by AByzhynar** [comment]
Reviewed

File: src/components/policy/src/cache_manager.cc [hunk 1]
```diff
@@ -1187,6 +1187,10 @@ bool CacheManager::IsPredataPolicy(const std::string& app_id) {
       pt_->policy_table.app_policies_section.apps[app_id];
 
   policy_table::Strings res;
+
+  sort(pre_data_app.groups.begin(), pre_data_app.groups.end());
```
Comment (LuxoftAKutsan) [inline src/components/policy/src/cache_manager.cc]: @orhan-mehmedov is it std:sort? 
Where did useing namespace used? We clarified that we should use it inly in functions scope. Please use std explicitly

File: src/components/policy/src/cache_manager.cc [hunk 2]
```diff
@@ -1187,6 +1187,10 @@ bool CacheManager::IsPredataPolicy(const std::string& app_id) {
...
+  sort(specific_app.groups.begin(), specific_app.groups.end());
+
   std::set_intersection(pre_data_app.groups.begin(),
```
Comment (anosach-luxoft) [inline src/components/policy/src/cache_manager.cc]: @orhan-mehmedov does not `set_intersection` work without sort?

Comment (orhan-mehmedov) [inline src/components/policy/src/cache_manager.cc]: @anosach-luxoft in windows platform there is an assert in the implementation of the standard library and is required to be sorted. By the way it is done in policy_helper.cc now I applied it here too.

File: src/components/policy/src/cache_manager.cc (hunk 1 above)
Comment (orhan-mehmedov) [inline src/components/policy/src/cache_manager.cc]: @LuxoftAKutsan fixed in [3d9672a](https://github.com/smartdevicelink/sdl_core/pull/794/commits/3d9672a458e8d7f73abde3ce47b8005395cbd63e)

**Comment by LuxoftAKutsan** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
Reviewed

**Pull request merged**